File: app/src/main/java/mk/webfactory/template/feature/home/ui/HomeContract.java
package mk.webfactory.template.feature.home.ui;

import mk.webfactory.template.feature.common.ui.BasePresenter;
import mk.webfactory.template.feature.common.ui.BaseView;

interface HomeContract {

  interface View extends BaseView<Presenter> {

    void showLoadingIndicator(boolean active);
  }

  abstract class Presenter extends BasePresenter<View> {

  }
}

File: app/src/main/java/mk/webfactory/template/feature/home/ui/mvp/HomeContract.kt
package mk.webfactory.template.feature.home.ui.mvp

import mk.webfactory.template.feature.common.ui.BasePresenter
import mk.webfactory.template.feature.common.ui.BaseView

interface HomeContract {
    interface View : BaseView<Presenter?> {
        fun showLoadingIndicator(active: Boolean)
    }

    abstract class Presenter :
        BasePresenter<View?>()
}